components/measure-upload/MeasureRepositoryUpload.tsx

```diff
@@ -0,0 +1,235 @@
+    if (id) {
+      const display = displayMap[id];
+      setIsLoadingPackage(PACKAGE_STATES.LOADING);
+      const response = await fetch(`${measureRepositoryUrl}/Measure/${id}/$package`, MEASURE_RETRIEVAL_OPTIONS);
+      const responseBody = await response.json();
+      if (responseBody.resourceType !== 'Bundle') {
+        if (responseBody.resourceType === 'OperationOutcome') {
+          rejectUpload(
+            `Packaging of Measure: ${display} failed with message: "${responseBody.issue[0].details.text}"`,
+            display
+          );
+        } else {
+          rejectUpload("Uploaded file must be a JSON FHIR Resource of type 'Bundle'", display);
+        }
+        return;
+      }
+
+      const measures = responseBody.entry.filter((e: fhir4.BundleEntry) => e.resource?.resourceType === 'Measure');
+      if (measures.length !== 1) {
+        rejectUpload(`Expected exactly 1 Measure in uploaded Bundle, received: ${measures.length}`, display);
+        return;
+      }
+
+      const missingValueSets = await identifyMissingValueSets(responseBody);
+      if (missingValueSets.length > 0) {
+        rejectUpload(missingValueSets, display, true);
+        return;
+      }
+      const effectivePeriod = ((measures as fhir4.BundleEntry[])[0].resource as fhir4.Measure).effectivePeriod;
+
+      // Set measurement period to default period
+      const measurementPeriod = populateMeasurementPeriod(effectivePeriod?.start, effectivePeriod?.end);
+      setMeasureBundle(mb => ({ ...mb, content: responseBody, isFile: false }));
+      setMeasurementPeriod(measurementPeriod);
+      setIsLoadingPackage(PACKAGE_STATES.SUCCESS);
+    } else {
+      setIsLoadingPackage(PACKAGE_STATES.NONE);
+    }
+  }
+
+  return (
+    <div>
+      <Group position="apart" style={{ alignItems: 'flex-end' }}>
+        <TextInput
+          label="Measure Repository Service URL"
+          placeholder="http://example.com"
+          value={measureRepositoryUrl}
+          onChange={handleMrsUrlChange}
+          style={{ flex: 1 }}
+        />
+        <Button onClick={retrieveMeasures} loading={isLoadingIds}>
+          Get Measures
+        </Button>
+      </Group>
+      <Collapse in={measureSelectIsOpen || (selectedMeasureId !== null && measureRepositoryUrl !== '')}>
+        <Grid>
+          <Grid.Col span={11}>
+            <Select
+              label="Measure ID"
+              data={idSelectData}
+              disabled={idSelectData.length === 0}
+              value={selectedMeasureId}
+              onChange={retrieveMeasurePackage}
+              placeholder="Select ID"
+              searchable
+              clearable
+              nothingFound={'No options matching search'}
+            />
+          </Grid.Col>
+          <Grid.Col span={1} style={{ display: 'flex', alignItems: 'end', justifyContent: 'center' }}>
+            {isLoadingPackage === PACKAGE_STATES.LOADING && <Loader />}
```

Comment:
Not sure how much overhead this would be, but I think it would be helpful to add some text alongside the loader saying that the measure is being packaged. When I had no ValueSets cached, the loader was spinning for a while, and so it can be unclear to the user that something is actually occurring under the hood. Then, when successful, we could add some text alongside the `CircleCheck` that says something along the lines of "Measure successfully packaged"